plutus-tx-plugin/test/AsData/Budget/9.6/patternMatching.uplc.golden

```diff
@@ -35,10 +35,10 @@ program
                                   (addInteger cse))
                                (addInteger cse))
                             (addInteger cse))
-                         (case cse [(\x y z w -> w)]))
+                         (case cse [(\x y z w -> x)]))
                       (case cse [(\x y z w -> y)]))
                    (case cse [(\x y z w -> z)]))
-                (case cse [(\x y z w -> x)]))
+                (case cse [(\x y z w -> w)]))
```

Comment:
This reordering bothers me a bit. Are we sure it's safe? I think it is, but it would be good to make sure. @zliu41 what do you think?

Reply:
It is safe in this instance, but it is weird, because the new transformation has no other effect on this test case except the reordering. If there's no opportunity for `ForceApply`, shouldn't it be a no-op?

@ana-pantilie do you have a guess for the reason of the reordering? Maybe you can print the programs before and after the ForceApply transformation and see how it changed.

Reply:
@zliu41 @effectfully I've pinned it down to being an interaction between `floatDelay`, `inline` and this optimisation. What I mean is that I have these results:
```
only old forceDelay
vs.
only new forceDelay
==> no change

old forceDelay && inline
vs.
new forceDelay && inline
==> no change

old forceDelay && floatDelay
vs.
new forceDelay && floatDelay
==> no change

old forceDelay && floatDelay && inline
vs
new forceDelay && floatDelay && inline
==> the exact change you see in this PR's diff
```

I didn't mess around with the order, it's always `floatDelay >=> forceDelay >=> ... >=> inline`.

I don't understand why, though, do you have any idea what might be happening? In any case, I can look into this more deeply tomorrow.

Reply:
Good investigation. Digging in further would be optional. I can imagine that `forceDelay` modifies the program in some way, but after `floatDelay` and `inline`, all that's left to be seen is the reordering.

If you do want to investigate further, you can try printing the program before and after `forceDelay`, and see what exactly is changed by `forceDelay`. One easy way to make a `term :: Term name uni fun ann` printable is to do `unsafeCoerce (void term) :: Term Name DefaultUni DefaultFun ()`, which can then be printed via `display . AsReadable`.

Reply:
Cool, thanks for the trick! I am having some trouble setting things up to just run my test, though, because this code runs at compile time and I get tons of unrelated output from other tests I believe. I'm going to merge this in the meantime.